src/main/java/frc/robot/led/LED.java

```diff
@@ -0,0 +1,77 @@
+package frc.robot.led;
+
+import static frc.robot.led.LEDConstants.BLACK;
+import static frc.robot.led.LEDConstants.MAX_HUE;
+import static frc.robot.led.LEDConstants.RAINBOW_INDICATOR;
+import static frc.robot.led.LEDConstants.RAINBOW_SATURATION;
+import static frc.robot.led.LEDConstants.RAINBOW_VALUE;
+import static frc.robot.led.LEDConstants.STRIP_LENGTH;
+
+import edu.wpi.first.wpilibj.util.Color;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class LED extends SubsystemBase {
+
+    private Color currentColor;
+
+    private final LEDComponents components;
+
+    private LED(LEDComponents components) {
+        this.components = components;
+    }
+
+    public String getCurrentColor() {
+        if (currentColor == null)
+            return "no color";
+        return "red: " + currentColor.red +
+                " green: " + currentColor.green +
+                " blue" + currentColor.blue;
+    }
+
+    private void setOneLed(int ledIndex, Color color) {
+        components.getBuffer().setLED(ledIndex, color);
+    }
+
+    public void setStrip(Color color) {
```

Comment:
If you want, I would recommand you creating a more generic setStrip, which would take also startingPos and endPos

This function will still exist, but it would use the more generic with starting pos as 0 and ledIndex as striplength